File: backend/app/domain/ticket_workflow.py
from app.models.ticket import TicketStatus

ALLOWED_TRANSITIONS = {
    TicketStatus.OPEN: [TicketStatus.IN_PROGRESS],
    TicketStatus.IN_PROGRESS: [TicketStatus.RESOLVED],
    TicketStatus.RESOLVED: [
        TicketStatus.CLOSED,
        TicketStatus.IN_PROGRESS,
    ],
    TicketStatus.CLOSED: [
        TicketStatus.IN_PROGRESS,
    ],
}

def can_transition(
    current: TicketStatus,
    target: TicketStatus,
) -> bool:
    return target in ALLOWED_TRANSITIONS.get(current, [])

File: frontend/src/utils/ticketWorkflow.ts
import type { TicketStatus } from "../types/ticket";

export const ALLOWED_TRANSITIONS: Record<TicketStatus, TicketStatus[]> = {
  OPEN: ["IN_PROGRESS"],
  IN_PROGRESS: ["RESOLVED"],
  RESOLVED: ["CLOSED", "IN_PROGRESS"],
  CLOSED: ["IN_PROGRESS"],
};

export const getValidNextStates = (currentStatus: TicketStatus): TicketStatus[] => {
  return ALLOWED_TRANSITIONS[currentStatus] || [];
};
